numba/cpython/unicode.py

```diff
@@ -2341,6 +2341,27 @@ def impl(i):
             return _PyUnicode_FromOrdinal(i)
         return impl
 
+
+@overload_method(types.Integer, '__str__')
+@overload_method(types.Integer, '__repr__')
+def integer_str(n):
+    def impl(n):
+        s = ''
+        flag = False
+        if n < 0:
+            n = -n
+            flag = True
+        if n == 0:
+            return '0'
+        while n > 0:
+            # there is an implicit conversion of n to float here
+            c = chr(int(ord('0') + (n % 10)))
+            n = n // 10
+            s = c + s
+        if flag:
+            s = '-' + s
+        return s
```

Comment:
Can we append to a `list` and then do `''.join(l)`. Your solution is `O(n^2)` I think, where `n` is the number of digits.

Reply:
You're right. I forgot this detail.

Reply:
A couple of notes: 
1) List is not supported inside `overload_method`. 
2) Since the largest value will be `max(uint64_t) = 2^64 = 18446744073709551616 => 20 digits`, a `O(n^2)` solution will have almost the same performance as a `O(n)` for `n=20`. If the current solution is not ideal, I can try to scratch something using typed list.

Reply:
In that case I'm okay with this solution.

Reply:
list is supported inside `overload_method`, what's not working?

Reply:
Am I missing something or is the suggestion here to do this:
```python
@overload_method(types.Integer, '__str__')
@overload_method(types.Integer, '__repr__')
def integer_str(n):
    def impl(n):
        s = []
        flag = False
        if n < 0:
            n = -n
            flag = True
        if n == 0:
            return '0'
        while n > 0:
            # there is an implicit conversion of n to float here
            c = chr(int(ord('0') + (n % 10)))
            n = n // 10
            s.append(c)
        if flag:
            s.append('-')
        return ''.join(s)
    return impl
```
or similar?

Reply:
oh it probably needs a reverse in the join, but the essence is there.

Reply:
You're right. I might have typed something wrong and the code didn't compile.